src/bootstrap/compile.rs

```diff
@@ -714,14 +721,21 @@ impl Step for Rustc {
             }
         }
 
+        for krate in &*self.crates {
+            cargo.arg("-p").arg(krate);
+        }
+
         builder.info(&format!(
-            "Building stage{} compiler artifacts ({} -> {})",
-            compiler.stage, &compiler.host, target
+            "Building{} stage{} compiler artifacts ({} -> {})",
+            crate_description(&self.crates),
+            compiler.stage,
+            &compiler.host,
+            target,
         ));
         run_cargo(
             builder,
             cargo,
-            self.crates.to_vec(),
+            vec![],
```

Comment:
nit (can be a future cleanup): Do we need the `tail_args` argument here anymore? It seems like ~any use of it should be replaceable with cargo.arg, right?

(Mainly asking to confirm that the change from passing via tail args to via cargo.arg is correct)

Reply:
Yes, that sounds right :) the only difference is which order the argument are passed, but I don't think there's ever a case where the order matters.

Reply:
done in https://github.com/rust-lang/rust/pull/106305